src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -39,169 +37,178 @@ public static async Task Main(string[] args)
                 return;
             }
 
-            CancellationTokenSource cts = new CancellationTokenSource();
+            var cts = new CancellationTokenSource();
             Console.CancelKeyPress +=
                 (sender, e) =>
                 {
                     e.Cancel = true;
                     cts.Cancel();
                 };
 
-            // QueryVisualStudioInstances returns Visual Studio installations on .NET Framework, and .NET Core SDK
-            // installations on .NET Core. We use the one with the most recent version.
-            var msBuildInstance = MSBuildLocator.QueryVisualStudioInstances().OrderByDescending(x => x.Version).First();
+            var cancellationToken = cts.Token;
+
+            var stopwatch = PerformanceTracker.StartNew();
 
-#if NETCOREAPP
-            // Since we do not inherit msbuild.deps.json when referencing the SDK copy
-            // of MSBuild and because the SDK no longer ships with version matched assemblies, we
-            // register an assembly loader that will load assemblies from the msbuild path with
-            // equal or higher version numbers than requested.
-            LooseVersionAssemblyLoader.Register(msBuildInstance.MSBuildPath);
-#endif
+            if (!string.IsNullOrEmpty(options.ProfileRoot))
+            {
+                ProfileOptimization.StartProfile(nameof(MSBuildWorkspace.OpenSolutionAsync));
+            }
 
-            MSBuildLocator.RegisterInstance(msBuildInstance);
+            using var workspace = await AnalyzerRunnerHelper.LoadSolutionAsync(options.SolutionPath, cancellationToken).ConfigureAwait(false);
+
+            foreach (var workspaceDiagnostic in workspace.Diagnostics)
+            {
+                if (workspaceDiagnostic.Kind == WorkspaceDiagnosticKind.Failure)
+                {
+                    Console.WriteLine(workspaceDiagnostic.Message);
+                }
+            }
+
+            Console.WriteLine($"Loaded solution in {stopwatch.GetSummary(preciseMemory: true)}");
+
+            var incrementalAnalyzerRunner = new IncrementalAnalyzerRunner(workspace, options);
+            var diagnosticAnalyzerRunner = new DiagnosticAnalyzerRunner(workspace.CurrentSolution, options);
+            var codeRefactoringRunner = new CodeRefactoringRunner(workspace, options);
 
-            var incrementalAnalyzerRunner = new IncrementalAnalyzerRunner(options);
-            var diagnosticAnalyzerRunner = new DiagnosticAnalyzerRunner(options);
-            var codeRefactoringRunner = new CodeRefactoringRunner(options);
             if (!incrementalAnalyzerRunner.HasAnalyzers && !diagnosticAnalyzerRunner.HasAnalyzers && !codeRefactoringRunner.HasRefactorings)
             {
                 WriteLine("No analyzers found", ConsoleColor.Red);
                 PrintHelp();
                 return;
             }
```

Comment:
❗ This is a frequent mistake by users. Solution load takes 45+ seconds for Roslyn; we don't want to delay this error until after the solution loads.

Reply:
Fixed